doc/cluster_design.md

```diff
@@ -0,0 +1,61 @@
+# Design: Clustering in SQLflow to analyze patterns in data
+
+## Concept
+
+For analysts and real business people, in the daily analysis work, most of the work is not prediction, but analysis of the patterns in the data. This can help them mine user behavioral characteristics and differences, helping the business discover value and operate.
+
+This design doc introduces how to support the `Cluster Model` in SQLFlow. 
+
+## User interface
+
+Users usually use a **TRAIN SQL** to train a model in Supervised learning. But, in this scenario, we focus on the extraction of data patterns in unsupervised learning. Therefore, we use **EXTRCT SQL** for pattern extraction, the simple pipeline like:
+
+EXTRCT SQL:
+
+``` sql
+SELECT * FROM train_table
+EXTRCT clusterModel
+WITH
+    model.encode_units = [100, 7]
+    model.n_clusters = 5
+COLUMN m1, m2, m3, m4, m5, m6, m7, m8, m9, m10 
+INTO my_cluster_model, result_table;
+```
+
+PREDICT SQL:
+``` sql
+SELECT *
+FROM new_table
+PREDICT result_test_table
+USING my_cluster_model;
+```
+
+where:
+- `train_table` is the high-dimensional table to be clustered.
+- `model.encode_units` is the autoencoder model layer's encoder units, the decode_units can reverse encode_units directly.
+- `model.n_clusters` is the number of patterns after clustering.
+- `my_cluster_model` is the trained cluster model.
+- `result_table` is the cluster result for train_table.
+- `new_table`: If you want to apply the model that extracts from train_table to the new data new_table directly, you can use **PREDICT SQL**. Note that the structure of new_table is the same as train_table, namely, same feature column.
+- `result_test_table` is the cluster result for new_table.
+
+## Implement Details
+-
+- 
+- 
+
+## Note
+- The **EXTRCT SQL** includes two models, the autoencode model and the cluster model. 
```

Comment:
Maybe we need a more general target for the keyword **EXTRACT**

Reply:
This keyword is just for illustration of the design. Next, maybe we can think about a better keywords together.